File: Algorithms/LinearSearch.cpp
#include<iostream>
using namespace std;

int linearSearch(int arr[], int size, int k) {
    for(int i=0; i<size; i++) {
        if(arr[i] == k) return i;

    }
    return -1;
}

int main() {
    int arr[10] = {1, 5, -7, 4, 3, 6, 9, 98, 35, -29};

    // Linear search finds if a target element exists in given array and returns its index. If it does not exist then it returns -1;
    int search10 = linearSearch(arr, 10, 10); // return -1
    int search98 = linearSearch(arr, 10, 98); // return 7

    return 0; 
}

File: Algorithms/linearsearch.java
public class LinearSearch {
    public static void main(String[] args) {
        // Array of numbers to search from
        int[] numbers = {12, 45, 67, 23, 89, 14, 56};

        // The target number we are searching for
        int target = 23;

        // Call the findNumber function and store the result (index of the target)
        int result = findNumber(numbers, target);

        // Check if the number was found and display the result
        if (result != -1) {
            System.out.println("Number " + target + " found at index: " + result);
        } else {
            System.out.println("Number " + target + " not found in the list.");
        }
    }

    // Function to perform a linear search in the array
    public static int findNumber(int[] list, int valueToFind) {
        // Loop through the array to search for the value
        for (int index = 0; index < list.length; index++) {
            // Check if the current element matches the value we're looking for
            if (list[index] == valueToFind) {
                return index; // Return the index if found
            }
        }
        return -1; // Return -1 if the value is not found in the array
    }
}
